### ISSUE TITLE
[Win11] mistral 7B performance down between 0.1.28 and 0.1.27

### ISSUE THREAD

**Issue by stevengans** [ollama/ollama#2960]
Using:
`curl http://localhost:11434/api/chat -d '{
  "model": "mistral",
  "messages": [
    { "role": "user", "content": "why is the sky blue?" }
  ]
}'`

Hardware: 
Gpu: Nvidia RTX A5000
Cpu: Intel i5-12600K
Mem: 64GB
Os: Windows 11 21H2

Performance is x4 slower on calls from what I've witnessed.

**Comment by jmorganca** [comment]
Hi @stevengans would it be possible to share the server logs? That will tell if Ollama is indeed using your GPU properly. You can find that by clicking on the Ollama taskbar item -> View Logs -> server.log. Thanks so much!

**Comment by stevengans** [comment]
Tested on same loop:
[server 0.1.27.log](https://github.com/ollama/ollama/files/14515371/server.0.1.27.log)
[server 0.1.28.log](https://github.com/ollama/ollama/files/14515372/server.0.1.28.log)

server0.1.27.log total time: 3.21m
server0.1.28.log total time: 4.23m

**Comment by stevengans** [comment]
I'll add that 0.1.27's logs are much nicer than 0.1.28...

**Comment by dhiltgen** [comment]
I just tried to repro on win 11 with a 4090 and perf seems relatively consistent for me.

0.1.27
```
PS C:\Users\danie> ollama run mistral
>>> /set verbose
Set 'verbose' mode.
>>> why is the sky blue?
 The color of the sky appears blue due to a phenomenon called Rayleigh scattering. As sunlight passes through
Earth's atmosphere, it interacts with different gases and particles present in the air. The short-wavelength blue
light gets scattered more easily than other colors because molecules in the atmosphere are smaller than the
wavelength of blue light. As a result, when we look up at the sky, we primarily see the scattered blue light,
making it appear blue to our eyes. However, during sunrise or sunset, the sky can display various shades of red,
pink, orange, and purple due to the presence of other gases like ozone and scattering at longer wavelengths.

total duration:       1.4990532s
load duration:        502µs
prompt eval count:    15 token(s)
prompt eval duration: 174.283ms
prompt eval rate:     86.07 tokens/s
eval count:           143 token(s)
eval duration:        1.323756s
eval rate:            108.03 tokens/s
>>> /bye
```

0.1.28:
```
PS C:\Users\danie> ollama run mistral
>>> /set verbose
Set 'verbose' mode.
>>> why is the sky blue?
 The color of the sky appears blue due to a natural phenomenon called Rayleigh scattering. As sunlight reaches
Earth's atmosphere, it interacts with gases and particles in the air, causing the shorter wavelengths of light
(blue and violet) to be scattered more effectively than longer wavelengths (yellow, orange, and red). Our eyes are
more sensitive to blue light, and we see the sky as blue during a clear day. However, it may appear other colors
depending on the presence or absence of other pollutants or particles in the atmosphere.

total duration:       1.3887087s
load duration:        518.3µs
prompt eval count:    15 token(s)
prompt eval duration: 224.925ms
prompt eval rate:     66.69 tokens/s
eval count:           117 token(s)
eval duration:        1.162431s
eval rate:            100.65 tokens/s
>>> /bye
```

With my setup, the model fully loads into the GPU, whereas on the RTX A5000 it's only able to load 26/33 layers to GPU, so perhaps there's a perf regression when models are split between CPU and GPU?

@stevengans can you try another model that fully fits in your GPU to see if you see the same perf hit, or if things are roughly the same?  I'll try some other models on my side as well to see if I can repro with a different approach.

**Comment by dhiltgen** [comment]
I just tried a larger model on my GPU: mixtral -> 27/33 layers to GPU

Perf actually got slightly better (rounding error though)

0.1.27
```
>>> why is the sky blue?
 The sky appears blue because of a process called Rayleigh scattering. As sunlight reaches Earth's atmosphere, it
is made up of different colors, which are essentially different wavelengths of light. Shorter wavelengths (like
violet and blue) are scattered in all directions more than other colors like red, orange, and yellow, which have
longer wavelengths.

Even though violet light is scattered more than blue light, the sky appears blue rather than violet because our
eyes are more sensitive to blue light and because sunlight reaches us with less violet light to begin with.
Additionally, some of the violet light gets absorbed by the ozone layer in the atmosphere. As a result, the sky we
observe is predominantly blue.

total duration:       8.9477387s
load duration:        504.6µs
prompt eval count:    16 token(s)
prompt eval duration: 581.987ms
prompt eval rate:     27.49 tokens/s
eval count:           154 token(s)
eval duration:        8.361723s
eval rate:            18.42 tokens/s
>>>
```

0.1.28
```
>>> why is the sky blue?
 The sky appears blue because of a process called Rayleigh scattering. As sunlight passes through the Earth's
atmosphere, it interacts with molecules and particles in the air such as nitrogen and oxygen. These particles
scatter the light in all directions. Blue light has a shorter wavelength and gets scattered more easily than other
colors, such as red or yellow, which have longer wavelengths. As a result, when we look up at the sky, we
predominantly see the blue light that has been scattered, making the sky appear blue.

total duration:       6.3294156s
load duration:        1.0415ms
prompt eval count:    16 token(s)
prompt eval duration: 605.577ms
prompt eval rate:     26.42 tokens/s
eval count:           111 token(s)
eval duration:        5.72223s
eval rate:            19.40 tokens/s
>>>
```

**Comment by stevengans** [comment]
Testing on mixtral:8x7b-instruct-v0.1-q2_K to see if it fits on my GPU. Will report back.

**Comment by stevengans** [comment]
@dhiltgen 
Ran the test for 0.1.28 (will now run the test for 0.1.27) but hit another issue below:
`llm_load_tensors: offloaded 33/33 layers to GPU
llm_load_tensors:        CPU buffer size =    41.02 MiB
llm_load_tensors:      CUDA0 buffer size = 14877.55 MiB`

It goes from working to hanging:

`{"function":"print_timings","level":"INFO","line":278,"msg":"generation eval time =    3584.46 ms /   162 runs   (   22.13 ms per token,    45.20 tokens per second)","n_decoded":162,"n_tokens_second":45.19511736502422,"slot_id":0,"t_token":22.126283950617285,"t_token_generation":3584.458,"task_id":199424,"tid":"3744","timestamp":1709815679}
{"function":"print_timings","level":"INFO","line":287,"msg":"          total time =    4034.65 ms","slot_id":0,"t_prompt_processing":450.195,"t_token_generation":3584.458,"t_total":4034.6530000000002,"task_id":199424,"tid":"3744","timestamp":1709815679}
{"function":"update_slots","level":"INFO","line":1635,"msg":"slot released","n_cache_tokens":1285,"n_ctx":2048,"n_past":1284,"n_system_tokens":0,"slot_id":0,"task_id":199424,"tid":"3744","timestamp":1709815679,"truncated":false}
[GIN] 2024/03/07 - 07:47:59 | 200 |    4.0380037s |       127.0.0.1 | POST     "/api/chat"
{"function":"launch_slot_with_data","level":"INFO","line":826,"msg":"slot is processing task","slot_id":0,"task_id":199589,"tid":"3744","timestamp":1709815679}
{"function":"update_slots","level":"INFO","line":1801,"msg":"slot progression","n_past":1052,"n_prompt_tokens_processed":77,"slot_id":0,"task_id":199589,"tid":"3744","timestamp":1709815679}
{"function":"update_slots","level":"INFO","line":1825,"msg":"kv cache rm [p0, end)","p0":1052,"slot_id":0,"task_id":199589,"tid":"3744","timestamp":1709815679}
{"function":"print_timings","level":"INFO","line":264,"msg":"prompt eval time     =     470.96 ms /    77 tokens (    6.12 ms per token,   163.50 tokens per second)","n_prompt_tokens_processed":77,"n_tokens_second":163.49653259950995,"slot_id":0,"t_prompt_processing":470.958,"t_token":6.116337662337663,"task_id":199589,"tid":"3744","timestamp":1709815690}
{"function":"print_timings","level":"INFO","line":278,"msg":"generation eval time =   10514.48 ms /   474 runs   (   22.18 ms per token,    45.08 tokens per second)","n_decoded":474,"n_tokens_second":45.0806844389181,"slot_id":0,"t_token":22.18244936708861,"t_token_generation":10514.481,"task_id":199589,"tid":"3744","timestamp":1709815690}
{"function":"print_timings","level":"INFO","line":287,"msg":"          total time =   10985.44 ms","slot_id":0,"t_prompt_processing":470.958,"t_token_generation":10514.481,"t_total":10985.439,"task_id":199589,"tid":"3744","timestamp":1709815690}
{"function":"update_slots","level":"INFO","line":1635,"msg":"slot released","n_cache_tokens":1603,"n_ctx":2048,"n_past":1602,"n_system_tokens":0,"slot_id":0,"task_id":199589,"tid":"3744","timestamp":1709815690,"truncated":false}
[GIN] 2024/03/07 - 07:48:10 | 200 |   10.9893584s |       127.0.0.1 | POST     "/api/chat"
{"function":"launch_slot_with_data","level":"INFO","line":826,"msg":"slot is processing task","slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815690}
{"function":"update_slots","level":"INFO","line":1801,"msg":"slot progression","n_past":1052,"n_prompt_tokens_processed":70,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815690}
{"function":"update_slots","level":"INFO","line":1825,"msg":"kv cache rm [p0, end)","p0":1052,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815690}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815712}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815738}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815763}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815787}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815810}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815833}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815861}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815884}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815906}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815929}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815952}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815974}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709815997}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816020}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816042}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816065}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816088}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816114}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816140}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816166}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816189}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816217}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816241}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816266}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816292}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816315}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816337}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816360}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816382}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816405}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816428}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816451}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816473}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816497}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816523}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816547}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816571}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816595}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816618}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816644}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816666}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816689}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816713}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816740}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816762}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816785}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816808}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816831}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816854}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816883}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816906}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816929}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816952}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709816975}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817000}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817023}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817046}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817068}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817091}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817114}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817136}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817160}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817188}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817211}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817236}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817263}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817293}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817317}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817340}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817369}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817394}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817417}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817440}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817463}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817485}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817510}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817534}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817561}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817587}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817613}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817636}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817659}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817682}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817710}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817735}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817758}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817780}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817804}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817827}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817850}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817873}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817895}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817918}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817941}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817964}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709817987}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709818010}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709818033}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709818056}
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":2048,"n_ctx":2048,"n_discard":1023,"n_keep":1,"n_left":2046,"n_past":2047,"n_system_tokens":0,"slot_id":0,"task_id":200066,"tid":"3744","timestamp":1709818082}`

**Comment by Pat02** [comment]
I am getting something very similar, when I run 'ollama run mistral' to test it works well for a few minutes, but quickly it just hangs and I never get the ollama prompt '>>>' , and the chat endpoint also hangs.

Hardware:
Gpu: Nvidia 4090
Cpu: Intel i9-14900K
Mem: 128GB
Os: Ubuntu Server 22.04.4 LTS

**Comment by stevengans** [comment]
@Pat02 I'm testing on RTX A5000. Can you run the test to see if this is the case on 0.1.27?

**Comment by Pat02** [comment]
is this what you mean by run the test?

/$ ollama run mistral
>>> /set verbose
Set 'verbose' mode.
>>> why is the sky blue?
 The color of the sky appears blue due to a process called scattering of light by the atmosphere. When the sun emits light, it contains all colors of the visible light spectrum in different intensities. However, as this
light travels through the Earth's atmosphere, it interacts with molecules such as nitrogen and oxygen, which scatter shorter wavelengths (blue and violet) more than longer wavelengths (red, green, and yellow).

As a result, when we look up at the sky, we primarily see blue light because it is scattered in all directions and reaches our eyes from every direction. However, during sunrise or sunset, the sky can appear red, pink, 
orange, or purple due to the different angles of the sun's rays interacting with the atmosphere and the presence of other pollutants or particles that scatter different wavelengths more effectively.

total duration:       1.309652804s
load duration:        1.025736ms
prompt eval duration: 17.495ms
prompt eval rate:     0.00 tokens/s
eval count:           184 token(s)
eval duration:        1.289725s
eval rate:            142.67 tokens/s

**Comment by batteryshark** [comment]
I'm having a similar issue via serve where it's blowing out my context... like after a few separate generate calls, it starts sending 
```
{"function":"update_slots","level":"INFO","line":1598,"msg":"slot context shift","n_cache_tokens":8192,"n_ctx":8192,"n_discard":4095,"n_keep":1,"n_left":8190,"n_past":8191,"n_system_tokens":0,"slot_id":0,"task_id":4194,"tid":"13692","timestamp":1709864519}
```

and either returns without understanding the context of what I gave it (which should fit fine in the context window) or it doesn't return at all and just spams this forever.

**Comment by stevengans** [comment]
@jmorganca There is a lot of interesting points here, but I think a new bug which has been encountered by @batteryshark @Pat02 and myself of a hang has been reproducible, whereas the initial bug I opened @dhiltgen hasn't been able to replicate and I haven't been able to as well...my guess is maybe I installed it badly? Would you be ok with closing this issue and then we can open a new issue for the above hang? Or would you want to keep this open?

**Comment by dhiltgen** [comment]
On the general topic of hanging, once #3218 is merged, we should be able to detect and recover from these failures.

**Comment by dhiltgen** [comment]
I'm going to close this one based on the original issue.

Release 0.1.32 will switch us over to subprocessing, which will enable us to detect and recover from hangs in the underlying llama.cpp code.
